cardano-testnet/src/Testnet/Runtime.hs

```diff
@@ -220,4 +256,83 @@ startLedgerNewEpochStateLogging testnetRuntime tmpWorkspace = withFrozenCallStac
             <> displayException e <> "\n"
           pure ConditionMet
 
+-- TODO: I hate this. Is there a better way to do this without
+-- reaching for concurrency primitives?
+isFileReadableLoop :: FilePath -> IO Bool
+isFileReadableLoop fp = do
+  threadDelay 100000
+  isFileReadable fp >>= \case
+    True -> return True
+    False -> isFileReadableLoop fp
+
+isFileReadable :: FilePath -> IO Bool
+isFileReadable fp = IO.withFile fp IO.ReadMode $ \h -> IO.hIsReadable h
+
+-- | Produce tuples that represent the change of the 'NewEpochState' after
+-- a transition.
+epochStateBeforeAfterValues
+  :: String
+  -> [(Text, Text)]
+epochStateBeforeAfterValues logFileContents =
+  let allEpochStates = filter (/= "") . Text.splitOn "#### BLOCK ####" $ Text.pack logFileContents
```

Comment:
Suggestion: move the block tag to a variable and allow reusing it. Both usage places are 40 lines apart, so it'd be easy to get overlooked by someone making changes here.